### PR TITLE
planner: generate wrong plan when update has subquery

### LINKED ISSUE #25658: Return wrong result when UPDATE's assignment_list has subquery
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```
CREATE TABLE `users` (
  `id` bigint(20) unsigned NOT NULL,
  `name` longtext DEFAULT NULL,
  `age` bigint(20) unsigned DEFAULT NULL,
  `company_id` bigint(20) DEFAULT NULL,
  `manager_id` bigint(20) unsigned DEFAULT NULL,
  `active` tinyint(1) DEFAULT NULL,
  `created_at` datetime(3) DEFAULT NULL,
  `updated_at` datetime(3) DEFAULT NULL,
  `deleted_at` datetime(3) DEFAULT NULL,
  PRIMARY KEY (`id`),
  KEY `idx_users_deleted_at` (`deleted_at`)
);
insert into users(id, company_id, name, updated_at) values(239, 15, 'Company 2 id 15', '2021-03-01 12:12:12.987');
create table companies(id bigint primary key, name longtext default null);
insert into companies values(14, 'Company 1 id 14');
insert into companies values(15, 'Company 2 id 15');
UPDATE `users` SET `company_id`=14,`name`=(SELECT `name` FROM `companies` WHERE companies.id = users.company_id),`updated_at`='2021-04-27 10:28:31.459' WHERE `id` = 239;
```

### 2. What did you expect to see? (Required)

```
select * from users;
```

name is not null
### 3. What did you see instead (Required)

but it is null
### 4. What is your TiDB version? (Required)

master branch

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #25658  

Problem Summary:

### What is changed and how it works?

if finding the subquery in the UPDATE's assignment list, the point_get return null.

What's Changed:

How it Works:

it change plan. the plan is shown below.
image

### Check List 

Tests 
- Manual test 

```
CREATE TABLE `users` (
  `id` bigint(20) unsigned NOT NULL,
  `name` longtext DEFAULT NULL,
  `age` bigint(20) unsigned DEFAULT NULL,
  `company_id` bigint(20) DEFAULT NULL,
  `manager_id` bigint(20) unsigned DEFAULT NULL,
  `active` tinyint(1) DEFAULT NULL,
  `created_at` datetime(3) DEFAULT NULL,
  `updated_at` datetime(3) DEFAULT NULL,
  `deleted_at` datetime(3) DEFAULT NULL,
  PRIMARY KEY (`id`),
  KEY `idx_users_deleted_at` (`deleted_at`)
);
insert into users(id, company_id, name, updated_at) values(239, 15, 'Company 2 id 15', '2021-03-01 12:12:12.987');
create table companies(id bigint primary key, name longtext default null);
insert into companies values(14, 'Company 1 id 14');
insert into companies values(15, 'Company 2 id 15');
UPDATE `users` SET `company_id`=14,`name`=(SELECT `name` FROM `companies` WHERE companies.id = users.company_id),`updated_at`='2021-04-27 10:28:31.459' WHERE `id` = 239;

```
Release note
- No release note

### CODE REVIEW

**Comment by tangenta** [comment]
Good job. A question: Is there any strong reason to apply a manual test instead of an integration test?

**Comment by bb7133** [comment]
Please add some test cases for your change.

File: planner/core/point_get_plan.go [hunk 1]
```diff
@@ -1280,6 +1280,11 @@ func findInPairs(colName string, pairs []nameValuePair) int {
 }
 
 func tryUpdatePointPlan(ctx sessionctx.Context, updateStmt *ast.UpdateStmt) Plan {
+	for _, list := range updateStmt.List {
+		if _, ok := list.Expr.(*ast.SubqueryExpr); ok {
```
Comment (AilinKid) [inline planner/core/point_get_plan.go:1284]: better add a comment about this? when update list contains subquery, avoid point get plan

Comment (hawkingrei) [reply planner/core/point_get_plan.go:1285]: Have added

**Comment by hawkingrei** [comment]
@bb7133 @tangenta @AilinKid  PTAL

**Review by bb7133 — APPROVED** [review]

**Comment by bb7133** [comment]
@hawkingrei Please try to check if #25658 can be reproduced in v4.0, v5.0 & v5.1, so that we will decide how many branches that backports would be needed.

**Comment by hawkingrei** [comment]
> @hawkingrei Please try to check if #25658 can be reproduced in v4.0, v5.0 & v5.1, so that we will decide how many branches that backports would be needed.

OK

**Pull request merged**